modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/internal/jaxrs/application/FDSApplication.java

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.data.set.taglib.internal.jaxrs.application;
+
+import com.liferay.frontend.data.set.FDSDataJSONFactory;
+import com.liferay.frontend.data.set.provider.FDSDataProvider;
+import com.liferay.frontend.data.set.provider.FDSDataProviderRegistry;
+import com.liferay.frontend.data.set.provider.search.FDSPagination;
+import com.liferay.frontend.data.set.taglib.internal.jaxrs.context.provider.PaginationContextProvider;
+import com.liferay.frontend.data.set.taglib.internal.jaxrs.context.provider.SortContextProvider;
+import com.liferay.frontend.data.set.taglib.internal.jaxrs.context.provider.ThemeDisplayContextProvider;
+import com.liferay.frontend.data.set.taglib.internal.servlet.ServletContextUtil;
+import com.liferay.frontend.taglib.clay.data.Filter;
+import com.liferay.frontend.taglib.clay.data.FilterFactory;
+import com.liferay.frontend.taglib.clay.data.FilterFactoryRegistry;
```

Comment:
These should be:
```java
com.liferay.frontend.data.set.provider.search.FDSKeywords;
com.liferay.frontend.data.set.provider.search.FDSKeywordsFactory;
com.liferay.frontend.data.set.provider.search.FDSKeywordsFactoryRegistry;
```
If you are wondering why we changed this so much: the naming of classes in `frontend.taglib.clay` made no sense for their purpose. [FDSFilter](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-api/src/main/java/com/liferay/frontend/data/set/filter/FDSFilter.java) is a model classes for UI component, looking misleadingly similar.

Reply:
Done! thanks for the explanation 🙂